### PR TITLE
HPCC4J-594 Add support for skipping tests

### PR DESCRIPTION
**Pull request by jpmcmu**
- Updated version of surefire plugin to support excludes file
- Added logic to generate an excludes file to GH actions

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [ ] I have created a corresponding JIRA ticket for this submission
- [ ] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by drealeed** [comment]
another issue is the ecl generation code that generates the files failing for some reason against the 160 and other clusters. Rodrigo had me comment it out locally in order to get my tests to run. Do you know if your changes will address that too?

**Comment by rpastrana** [comment]
Hi @drealeed those aren't necessarily related to this change, can you write up the executeECLScript problem? Thanks

File: dfsclient/pom.xml [hunk 1]
```diff
@@ -34,6 +34,7 @@
                 <value>org.hpccsystems.ws.client.TestResultNotifier</value>
               </property>
             </properties>
+            <excludesFile>../excluded-tests.txt</excludesFile>
```
Comment (rpastrana) [inline dfsclient/pom.xml:37]: it seems ok to have a single excluded-tests file at the hpcc4j level, but I could also see an argument for dedicated excluded-tests files per sub project.
what are the main drawbacks/advantages of the current implementation?

File: pom.xml [hunk 2]
```diff
@@ -45,7 +45,7 @@
     <maven.deploy.version>2.8.2</maven.deploy.version>
```
Comment (rpastrana) [inline pom.xml:45]: can we have a sample entry in the above excluded-tests.txt file?
Perhaps create a new test simply for the purpose of always excluding it via this file?
Also, does the github action mechanism append excluded tests entries, or does it overwrite the file?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu a few questions.

File: wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java [hunk 3]
```diff
@@ -6,9 +6,11 @@
 
 import org.hpccsystems.ws.client.wrappers.ArrayOfEspExceptionWrapper;
 import org.hpccsystems.ws.client.wrappers.wsdfu.DFUDataColumnWrapper;
+import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
+@Ignore
```
Comment (rpastrana) [inline wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java:13]: @drealeed I'd like to integrate these tests into the standard wsclient tests, is there anything RAMPS specific being tested? If so, could you provide ECL to recreate the structures being testsed here?

Comment (drealeed) [reply wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java:13]: No, this was just a way to get a really comprehensive test of all possible record structures; I picked ramps dev because it had a lot of files for the regression test and I had access to it.  And at the time the ecl errors occurred reading files from ramps dev so I pointed the test at the previously failing files to make sure they worked now.

Comment (drealeed) [reply wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java:13]: Unfortunately the files tested in testFileMetadata() no longer exist. That test can be removed. We have similar tests in tardis for different hpcc ecl structures now.

Comment (rpastrana) [reply wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java:13]: Unless those ecl structures are proprietary, I'd like to absorb as many tests as possible. Does it make sense to donate those tests given the fact we can preemptively create the target files via ecl scripts?

Comment (rpastrana) [reply wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java:13]: @drealeed do you recall how test::wsclient::noecl was created? it seems like an edge case we should continue to test
